---
title: "TADCompare"
author: "Matt"
date: "1/2022"
output:
  html_document: default
  pdf_document: default
---




Applying TADCompare (https://www.frontiersin.org/articles/10.3389/fgene.2020.00158/full) to a set of HiC contact matrices. I use matrices generated by HiCExplorer from all runs of NextSeq sequencing. The Hi-C contacts have been normalized and corrected with ICE. For TADCompare, the matrices should be .txt form cooler matrices, as explained in the vignette (https://bioconductor.org/packages/devel/bioc/vignettes/TADCompare/inst/doc/TADCompare.html). Do this using HiCExplorer's matrix convert function. The inputs here are the .txt form matrices to be compared.




```{r, initialize packages, results="hide"}
#necessary packages
library(dplyr)
library(TADCompare)
library(HiCcompare)
library(ggplot2)
library(pheatmap)

#library(SpectralTAD)       #they also mention SpectralTAD, which is another TAD calling option. I don't use it here.
```


```{r, read matrices for all repeats, results="hide", echo=FALSE}

#####setwd to wherever the matrices are, change this line
setwd("C:/Users/Owner/Desktop/Lab Work/NextSeq_all/TADcompare/")

#as before, read the matrices into R from files.
cool_matXO2 <- read.table("XO_2_ICE.txt")
cool_matXO3 <- read.table("XO_3_ICE.txt")
cool_matXO4 <- read.table("XO_4_ICE.txt")
cool_matXY2 <- read.table("XY_2_ICE.txt")
cool_matXY3 <- read.table("XY_3_ICE.txt")
cool_matXY4 <- read.table("XY_4_ICE.txt")
cool_matX_Y2 <- read.table("X_Y_2_ICE.txt")
cool_matX_Y3 <- read.table("X_Y_3_ICE.txt")
cool_matX_Y4 <- read.table("X_Y_4_ICE.txt")
cool_matX_YY2 <- read.table("X_YY_2_ICE.txt")
cool_matX_YY3 <- read.table("X_YY_3_ICE.txt")
cool_matX_YY4 <- read.table("X_YY_4_ICE.txt")

#get matrices into sparse form, required for TADCompare
sparse_matXO2 <- HiCcompare::cooler2sparse(cool_matXO2)
sparse_matXO3 <- HiCcompare::cooler2sparse(cool_matXO3)
sparse_matXO4 <- HiCcompare::cooler2sparse(cool_matXO4)
sparse_matXY2 <- HiCcompare::cooler2sparse(cool_matXY2)
sparse_matXY3 <- HiCcompare::cooler2sparse(cool_matXY3)
sparse_matXY4 <- HiCcompare::cooler2sparse(cool_matXY4)
sparse_matX_Y2 <- HiCcompare::cooler2sparse(cool_matX_Y2)
sparse_matX_Y3 <- HiCcompare::cooler2sparse(cool_matX_Y3)
sparse_matX_Y4 <- HiCcompare::cooler2sparse(cool_matX_Y4)
sparse_matX_YY2 <- HiCcompare::cooler2sparse(cool_matX_YY2)
sparse_matX_YY3 <- HiCcompare::cooler2sparse(cool_matX_YY3)
sparse_matX_YY4 <- HiCcompare::cooler2sparse(cool_matX_YY4)

#remove Y scaffolds.
XO_2_non_Y=c(sparse_matXO2["chrX"],sparse_matXO2["chr2L"],sparse_matXO2["chr2R"],sparse_matXO2["chr3L"],sparse_matXO2["chr3R"],sparse_matXO2["chr4"])
XO_3_non_Y=c(sparse_matXO3["chrX"],sparse_matXO3["chr2L"],sparse_matXO3["chr2R"],sparse_matXO3["chr3L"],sparse_matXO3["chr3R"],sparse_matXO3["chr4"])
XO_4_non_Y=c(sparse_matXO4["chrX"],sparse_matXO4["chr2L"],sparse_matXO4["chr2R"],sparse_matXO4["chr3L"],sparse_matXO4["chr3R"],sparse_matXO4["chr4"])
XY_2_non_Y=c(sparse_matXY2["chrX"],sparse_matXY2["chr2L"],sparse_matXY2["chr2R"],sparse_matXY2["chr3L"],sparse_matXY2["chr3R"],sparse_matXY2["chr4"])
XY_3_non_Y=c(sparse_matXY3["chrX"],sparse_matXY3["chr2L"],sparse_matXY3["chr2R"],sparse_matXY3["chr3L"],sparse_matXY3["chr3R"],sparse_matXY3["chr4"])
XY_4_non_Y=c(sparse_matXY4["chrX"],sparse_matXY4["chr2L"],sparse_matXY4["chr2R"],sparse_matXY4["chr3L"],sparse_matXY4["chr3R"],sparse_matXY4["chr4"])
X_Y_2_non_Y=c(sparse_matX_Y2["chrX"],sparse_matX_Y2["chr2L"],sparse_matX_Y2["chr2R"],sparse_matX_Y2["chr3L"],sparse_matX_Y2["chr3R"],sparse_matX_Y2["chr4"])
X_Y_3_non_Y=c(sparse_matX_Y3["chrX"],sparse_matX_Y3["chr2L"],sparse_matX_Y3["chr2R"],sparse_matX_Y3["chr3L"],sparse_matX_Y3["chr3R"],sparse_matX_Y3["chr4"])
X_Y_4_non_Y=c(sparse_matX_Y4["chrX"],sparse_matX_Y4["chr2L"],sparse_matX_Y4["chr2R"],sparse_matX_Y4["chr3L"],sparse_matX_Y4["chr3R"],sparse_matX_Y4["chr4"])
X_YY_2_non_Y=c(sparse_matX_YY2["chrX"],sparse_matX_YY2["chr2L"],sparse_matX_YY2["chr2R"],sparse_matX_YY2["chr3L"],sparse_matX_YY2["chr3R"],sparse_matX_YY2["chr4"])
X_YY_3_non_Y=c(sparse_matX_YY3["chrX"],sparse_matX_YY3["chr2L"],sparse_matX_YY3["chr2R"],sparse_matX_YY3["chr3L"],sparse_matX_YY3["chr3R"],sparse_matX_YY3["chr4"])
X_YY_4_non_Y=c(sparse_matX_YY4["chrX"],sparse_matX_YY4["chr2L"],sparse_matX_YY4["chr2R"],sparse_matX_YY4["chr3L"],sparse_matX_YY4["chr3R"],sparse_matX_YY4["chr4"])

#remove objects that won't be used
rm(cool_matXO2,cool_matXO3,cool_matXO4, sparse_matXO2, sparse_matXO3, sparse_matXO4, cool_matXY2, cool_matXY3, cool_matXY4, sparse_matXY2, sparse_matXY3, sparse_matXY4,cool_matX_Y2,cool_matX_Y3,cool_matX_Y4,sparse_matX_Y2,sparse_matX_Y3,sparse_matX_Y4,cool_matX_YY2,cool_matX_YY3,cool_matX_YY4,sparse_matX_YY2,
sparse_matX_YY3,sparse_matX_YY4)


```
With 12 matrices (reps 2,3,4 of genotypes XO, XY, X_Y, and X_YY), there are 66 comparisons to make. The next code block uses a loop to make all comparisons. It works, but it runs out of memory on my computer unless I remove the comparison objects after plotting. The full set of comparisons takes ~10 mins with 32 GB RAM. 

There are 66 labeled plots, one for each possible comparison between two data sets. The overall proportions of each boundary type are similar to those for the merged comparisons: most boundaries are non-differential, and most changes that occur are complex.

At a glance, I don't see striking differences when comparing within samples and between samples. What would be the best way to make a statistical comparison, for a more rigorous check of whether TAD boundary changes differ between samples compared to within samples?

Even though I don't detect obvious genome-wide differences in TADs, it's still possible that there are a few individual differential TADs that are biologically relevant. These would need to be identified another way.

Notes 
- Chr4 is ignored for all of these comparisons, since not many boundary changes are detected and the detected changes are not of all types. 
- If you're interested in seeing a specific comparison between any two replicates, run the comparison outside of a loop using the next code block. 



```{r, TADCompare for all repeats}

#start by setting the order of objects to compare
order = c("XO_2","XO_3","XO_4","XY_2","XY_3","XY_4","X_Y_2","X_Y_3","X_Y_4","X_YY_2","X_YY_3","X_YY_4")


#nested loops, comparing objects 2-12 with object 1, then 3-12 with object 2, and continuing.
for (i in order) {
  
  order = order[- 1]
  for (j in order){

    
    #set the name of the current comparison
    comparison_name = paste("Diff_tads_",i,".",j,sep="")
    
    
    #comparing all chromosomes and keeping naming consistent
    assign(comparison_name,lapply(names(get(paste(i,"non_Y",sep="_"))), function(x) {
      TADCompare(get(paste(i,"non_Y",sep="_"))[[x]], get(paste(j,"non_Y",sep="_"))[[x]], resolution = 5000)
      }))
    
    #merge the counts for chrs X, 2, 3
    merged_data= 
      get(comparison_name)[[1]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[2]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[3]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[4]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[5]][["Count_Plot"]][["data"]][["Count"]]
    
    #add the merged data to a plot and title with the comparison
    comparison_plot = get(comparison_name)[[2]][["Count_Plot"]]
    comparison_plot[["data"]][["Count"]] = merged_data
    comparison_plot = comparison_plot+ggtitle(paste(i,"vs.",j,sep=" "))
    
    print(comparison_plot)
    
  
    #remove files to save memory
    rm(merged_data,list=ls(pattern="Diff_tads"))

  }
  
}

#remove objects that are no longer needed
rm(comparison_plot,order,i,j,comparison_name)



```





Here's the code to plot a single comparison. This can be changed for any comparison of interest.
```{r, TADCompare for specific repeats, results="hide"}
#compare two lists of Hi-C Contacts


####CHANGE LIST_1 and LIST_2, and all placeholder object names
diff_tads_COMPARISON = lapply(names(XO_2_non_Y), function(x) {
  TADCompare(XO_2_non_Y[[x]], XY_4_non_Y[[x]], resolution = 5000)
})

#merge chrs X, 2, 3
merged_chrs_COMPARISON= 
      diff_tads_COMPARISON[[1]][["Count_Plot"]][["data"]][["Count"]]+
      diff_tads_COMPARISON[[2]][["Count_Plot"]][["data"]][["Count"]]+
      diff_tads_COMPARISON[[3]][["Count_Plot"]][["data"]][["Count"]]+
      diff_tads_COMPARISON[[4]][["Count_Plot"]][["data"]][["Count"]]+
      diff_tads_COMPARISON[[5]][["Count_Plot"]][["data"]][["Count"]]
    
    #add the merged data to a plot and title with the comparison
COMPARISON_plot = diff_tads_COMPARISON[[2]][["Count_Plot"]]
COMPARISON_plot[["data"]][["Count"]] = merged_chrs_COMPARISON
COMPARISON_plot = COMPARISON_plot+ggtitle("XO_2 vs. XO_3")

percent_ND_XO2_XY4=(merged_chrs_COMPARISON[3])/(merged_chrs_COMPARISON[1]+merged_chrs_COMPARISON[2]+merged_chrs_COMPARISON[3]+merged_chrs_COMPARISON[4]+merged_chrs_COMPARISON[5]+merged_chrs_COMPARISON[6])
    
print(COMPARISON_plot)
```

```{r, trying to make dendrogram, results="hide"}

percent_ND_matrix=matrix(nrow = 12, ncol = 12)
rownames(percent_ND_matrix)=c("XO_2","XO_3","XO_4","XY_2","XY_3","XY_4","X_Y_2","X_Y_3","X_Y_4","X_YY_2","X_YY_3","X_YY_4")
colnames(percent_ND_matrix)=c("XO_2","XO_3","XO_4","XY_2","XY_3","XY_4","X_Y_2","X_Y_3","X_Y_4","X_YY_2","X_YY_3","X_YY_4")

#start by setting the order of objects to compare
order = c("XO_2","XO_3","XO_4","XY_2","XY_3","XY_4","X_Y_2","X_Y_3","X_Y_4","X_YY_2","X_YY_3","X_YY_4")


percent_ND=as.data.frame(matrix(nrow=66,ncol=2))
colnames(percent_ND)=c("score","comparison")
#nested loops, comparing objects 2-12 with object 1, then 3-12 with object 2, and continuing.
for (i in order) {
  
  percent_ND_matrix[i,i]=NA
  
  order = order[- 1]
  for (j in order){

    
    #set the name of the current comparison
    comparison_name = paste("Diff_tads_",i,".",j,sep="")
    
    
    #comparing all chromosomes and keeping naming consistent
    assign(comparison_name,lapply(names(get(paste(i,"non_Y",sep="_"))), function(x) {
      TADCompare(get(paste(i,"non_Y",sep="_"))[[x]], get(paste(j,"non_Y",sep="_"))[[x]], resolution = 5000)
      }))
    
    #merge the counts for chrs X, 2, 3
    merged_data= 
      get(comparison_name)[[1]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[2]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[3]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[4]][["Count_Plot"]][["data"]][["Count"]]+
      get(comparison_name)[[5]][["Count_Plot"]][["data"]][["Count"]]
    
    percent_ND_matrix[i,j]=(merged_data[3])/(merged_data[1]+merged_data[2]+merged_data[3]+merged_data[4]+merged_data[5]+merged_data[6])
    percent_ND_matrix[j,i]=(merged_data[3])/(merged_data[1]+merged_data[2]+merged_data[3]+merged_data[4]+merged_data[5]+merged_data[6])
    
    
  
    #remove files to save memory
    rm(merged_data,list=ls(pattern="Diff_tads"))

  }
  
}

#remove objects that are no longer needed
rm(order,i,j,comparison_name)


```

```{r}

percent_ND_matrix[1,1]=NA
percent_ND_matrix[2,2]=NA
percent_ND_matrix[3,3]=NA
percent_ND_matrix[4,4]=NA
percent_ND_matrix[5,5]=NA
percent_ND_matrix[6,6]=NA
percent_ND_matrix[7,7]=NA
percent_ND_matrix[8,8]=NA
percent_ND_matrix[9,9]=NA
percent_ND_matrix[10,10]=NA
percent_ND_matrix[11,11]=NA
percent_ND_matrix[12,12]=NA

pheatmap(percent_ND_matrix)
pheatmap(percent_ND_matrix, cluster_rows=F, cluster_cols=F)

#heatmap(percent_ND_matrix, scale = "none")




```


This heatmap/dendrogram shows the percent of total TADs that are non-differential in each pairwise comparison between all samples. It seems to show no strong Y dosage effect, or really any clear effect. I still think it's possible that smaller-scale results are interesting - perhaps there's a TAD somewhere that correlates with Y dosage, but it isn't picked up when considering the ~2000 TADs total. Still, this result seems to indicate that Y dosage doesn't cause above-background levels of TAD border reshuffling - which is a result.









```{r}
#####setwd to wherever the matrices are, change this line
setwd("C:/Users/Owner/Desktop/Lab Work/NextSeq_all/TADcompare/")

#as before, read the matrices into R from files.
cool_matXO <- read.table("XO_merged_ICE.txt")
cool_matXY <- read.table("XY_merged_ICE.txt")
cool_matX_Y <- read.table("X_Y_merged_ICE.txt")
cool_matX_YY <- read.table("X_YY_merged_ICE.txt")


#get matrices into sparse form, required for TADCompare
sparse_matXO <- HiCcompare::cooler2sparse(cool_matXO)
sparse_matXY <- HiCcompare::cooler2sparse(cool_matXY)
sparse_matX_Y <- HiCcompare::cooler2sparse(cool_matX_Y)
sparse_matX_YY <- HiCcompare::cooler2sparse(cool_matX_YY)


#remove Y scaffolds.
XO_non_Y=c(sparse_matXO["chrX"],sparse_matXO["chr2L"],sparse_matXO["chr2R"],sparse_matXO["chr3L"],sparse_matXO["chr3R"],sparse_matXO["chr4"])
XY_non_Y=c(sparse_matXY["chrX"],sparse_matXY["chr2L"],sparse_matXY["chr2R"],sparse_matXY["chr3L"],sparse_matXY["chr3R"],sparse_matXY["chr4"])
X_Y_non_Y=c(sparse_matX_Y["chrX"],sparse_matX_Y["chr2L"],sparse_matX_Y["chr2R"],sparse_matX_Y["chr3L"],sparse_matX_Y["chr3R"],sparse_matX_Y["chr4"])
X_YY_non_Y=c(sparse_matX_YY["chrX"],sparse_matX_YY["chr2L"],sparse_matX_YY["chr2R"],sparse_matX_YY["chr3L"],sparse_matX_YY["chr3R"],sparse_matX_YY["chr4"])

#remove objects that won't be used
rm(cool_matXO, sparse_matXO, cool_matXY, sparse_matXY, cool_matX_Y, sparse_matX_Y, cool_matX_YY, sparse_matX_YY)
```




```{r}



#start by setting the order of objects to compare
order = c("XO","XY","X_Y","X_YY")


#nested loops, comparing objects 2-12 with object 1, then 3-12 with object 2, and continuing.
for (i in order) {
  
  
  order = order[- 1]
  for (j in order){

    
    #set the name of the current comparison
    comparison_name = paste(i,"_vs_",j,sep="")
    
    
    #comparing all chromosomes and keeping naming consistent
    assign(comparison_name,lapply(names(get(paste(i,"non_Y",sep="_"))), function(x) {
      TADCompare(get(paste(i,"non_Y",sep="_"))[[x]], get(paste(j,"non_Y",sep="_"))[[x]], resolution = 5000)
      }))
    


  }
  
}

#remove objects that are no longer needed
rm(order,i,j,comparison_name)


```





```{r}

high_value_test=max(XO_vs_XY[[1]]$Boundary_Scores$Boundary)



augmented_boundaries_test=XO_vs_XY[[2]]
augmented_boundaries_test$Boundary_Scores$Boundary=XO_vs_XY[[2]]$Boundary_Scores$Boundary+high_value_test


chromosomes=c("chrX","chr2L","chr2R","chr3L","chr3R","chr4")
comparisons=c("XO_vs_XY","XO_vs_X_Y","XO_vs_X_YY","XY_vs_X_Y","XY_vs_X_YY","X_Y_vs_X_YY")


objects=list(XO_vs_XY,XO_vs_X_Y,XO_vs_X_YY,XY_vs_X_Y,XY_vs_X_YY,X_Y_vs_X_YY)

for (j in 1:6){

  current=objects[[j]]
  df=list("X","2L","2R","3L","3R","4")
  highlight_df=list("X","2L","2R","3L","3R","4")

  high_value=0
  chr_positions = vector("numeric", 7)
  chr_positions[1]=0

  for (i in 1:6){
    df[[i]]=current[[i]]$Boundary_Scores
    df[[i]]$Boundary=df[[i]]$Boundary+high_value
    
    highlight_df[[i]] = df[[i]] %>% filter(abs(Gap_Score)>=5)
    
    
    
    high_value=max(df[[i]]$Boundary)
    chr_positions[i+1]=(high_value)
    
  }
  append=rbind(df[[1]], df[[2]], df[[3]], df[[4]], df[[5]], df[[6]])
  append_highlight=rbind(highlight_df[[1]], highlight_df[[2]], highlight_df[[3]], highlight_df[[4]], highlight_df[[5]], highlight_df[[6]])
  plot=ggplot(append,aes(Boundary,Gap_Score))+geom_point() + geom_point(data=append_highlight,color='blue') +ggtitle(paste(comparisons[j]))+coord_cartesian(ylim=c(-15,15)) + ylab("Differential Boundary Score") + xlab("Chromosome")+ scale_x_continuous(breaks=chr_positions, labels=c("chrX","chr2","","chr3","","chr4", ""))                                                                                                                                                     
  print(plot)
}



```















